### PR TITLE
Reduce dimension with random projection

### PR DESCRIPTION
**Pull request by KT12**
@dhimmel , I was able to branch and add the files.  Thanks for your guidance.

In response to [#71](https://github.com/cognoma/machine-learning/pull/71) and @htcai 's work.

This pull request incorporates [random projection](http://www.ime.unicamp.br/~wanderson/Artigos/randon_projection_kdd.pdf) to reduce the dimensionality of the data.

Succinct [PCA vs random projection comparison](http://stats.stackexchange.com/questions/235632/pca-vs-random-projection).

### CODE REVIEW

File: algorithms/scripts/SGDClassifier-RandomProjection-KT12.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+# Typically, this can only be done where the number of mutations is large enough
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
+'Size: {:,} features, {:,} training samples, {:,} testing samples'.format(len(X.columns), len(X_train), len(X_test))
+
+
+# ## Reduce the dimensionality via Random Projection
+
+# In[12]:
+
+transformer = GaussianRandomProjection() # num of components on auto
```
Comment (dhimmel) [inline algorithms/scripts/SGDClassifier-RandomProjection-KT12.py]: Let's make sure we're specifying `random_state` so the results are reproducible.

Comment (KT12) [reply algorithms/scripts/SGDClassifier-RandomProjection-KT12.py]: Thanks for catching that.  Will amend.

**Review by dhimmel — APPROVED** [review]

**Comment by dhimmel** [comment]
@KT12 any more changes you want to make or should I merge?

**Comment by KT12** [comment]
Please merge.  Thank you.

**Pull request merged**